src/internal/storage/fileset/reader.go

```diff
@@ -49,24 +48,36 @@ func (r *Reader) getPrimitive(ctx context.Context) (*Primitive, error) {
 	return prim, nil
 }
 
-func (r *Reader) IterateDeletes(ctx context.Context, cb func(File) error, opts ...index.Option) error {
+// withPrimitive retrieves a primitive and passes it in to the user supplied callback. It also spawns a child
+// context that is enriched with fields related to the primitive.
+func (r *Reader) withPrimitive(ctx context.Context, cb func(ctx context.Context, prim *Primitive) error) error {
+	ctx = pctx.Child(ctx, "reader")
 	prim, err := r.getPrimitive(ctx)
 	if err != nil {
 		return err
 	}
-	ir := index.NewReader(r.chunks, r.idxCache, prim.Deletive, opts...)
-	return ir.Iterate(ctx, func(idx *index.Index) error {
-		return cb(newFileReader(r.chunks, idx))
+	ctx = pctx.Child(ctx, "", pctx.WithFields(LogIndex(prim.Additive, "startIdx")))
```

Comment:
The start index should be recorded by the index reader imo. Doing at this level misses the deletive fileset.